dig.go

```diff
@@ -39,6 +39,20 @@ type key struct {
 	name string
 }
 
+// Option configures a Container. It's included for future functionality;
+// currently, there are no concrete implementations.
+type Option interface {
+	apply(*Container)
+}
+
+// A ProvideOption modifies the default behavior of Provide. It's included for
+// future functionality; currently, there are no concrete implementations.
+type ProvideOption interface{}
```

Comment:
should we have an actual private interface? Type aliasing to interface{} widens the DIG API too much :)